flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/operator/AbstractKeyedCEPPatternOperator.java

```diff
@@ -126,6 +129,7 @@ public AbstractKeyedCEPPatternOperator(
 		this.isProcessingTime = Preconditions.checkNotNull(isProcessingTime);
 		this.keySerializer = Preconditions.checkNotNull(keySerializer);
 		this.nfaFactory = Preconditions.checkNotNull(nfaFactory);
+		this.conditionRegistry = nfaFactory.createNFA().getConditionRegistry();
```

Comment:
Maybe let's access conditionRegistry through nfaFactory rather than NFA itselfs?

Anyway I think we should think of removing one layer in the `NFACompiler.java`, but I will create a seperate JIRA for this.